ocs_ci/deployment/hosted_cluster.py

```diff
@@ -1824,13 +1824,33 @@ def hypershift_cluster_factory(
                 ] = running_ocp_version
 
             cluster_path = create_cluster_dir(cluster_name)
-            kubeconf_path = (
+            logger.info("Cluster path: %s", cluster_path)
+            def_client_config_dict["ENV_DATA"]["cluster_path"] = cluster_path
+
+            kubeconf_paths = (
                 hosted_clients_obj.download_hosted_clusters_kubeconfig_files(
                     {cluster_name: cluster_path}
                 )
             )
+            if not len(kubeconf_paths):
+                from ocs_ci.framework.exceptions import ClusterKubeconfigNotFoundError
+
+                ClusterKubeconfigNotFoundError(
+                    f"Failed to download kubeconfig for cluster {cluster_name}"
+                )
+            else:
+                kubeconf_path = [
+                    path for path in kubeconf_paths if cluster_name in path
+                ][0]
```

Comment:
If I understand it correctly, this expects that the cluster name is used in the path for the kubeconfig, which we can probably assume, but there might be a problem with a situation when one cluster name will be a substring of another cluster name, right? For example `cluster-a` and `cluster-aa`.

Reply:
good point.
We can try to read kubeconfig.clusters.name
Kubeconfig has list of clusters and one of them have matching name, others are dedicated for routes